Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -580,6 +580,7 @@ - (instancetype)_initWithConfiguration:(_WKWebsiteDataStoreConfiguration *)confi
 
     auto sessionID = configuration.isPersistent ? PAL::SessionID::generatePersistentSessionID() : PAL::SessionID::generateEphemeralSessionID();
     API::Object::constructInWrapper<WebKit::WebsiteDataStore>(self, configuration->_configuration->copy(), sessionID);
+    _websiteDataStore->dispatchResolveDirectories();
```

Comment:
This should be in the constructor / factory function, like it used to be. Too likely to forget to call otherwise.

Reply:
We cannot create Ref for WebsiteDataStore in constructor because it's not RefCounted. It inherits APIObject which calls CFRetain for ref(). It would crash inside CFRetain with:
```
Thread 0 Crashed:
0   CoreFoundation                	       0x180e3df50 CFRetain.cold.1 + 16
1   CoreFoundation                	       0x180c5a6e0 CFRetain + 140
2   WebKit                        	       0x131a2814c API::Object::ref() const + 32 (APIObject.mm:126)
3   WebKit                        	       0x132a2fb9c WTF::DefaultRefDerefTraits<WebKit::WebsiteDataStore>::ref(WebKit::WebsiteDataStore&) + 28 (Ref.h:55)
4   WebKit                        	       0x132a2fb68 WTF::Ref<WebKit::WebsiteDataStore, WTF::RawPtrTraits<WebKit::WebsiteDataStore>, WTF::DefaultRefDerefTraits<WebKit::WebsiteDataStore>>::Ref(WebKit::WebsiteDataStore&) + 40 (Ref.h:86)
5   WebKit                        	       0x132972970 WTF::Ref<WebKit::WebsiteDataStore, WTF::RawPtrTraits<WebKit::WebsiteDataStore>, WTF::DefaultRefDerefTraits<WebKit::WebsiteDataStore>>::Ref(WebKit::WebsiteDataStore&) + 40 (Ref.h:87)
6   WebKit                        	       0x132e5f308 WebKit::WebsiteDataStore::dispatchResolveDirectories() + 200 (WebsiteDataStore.cpp:472)
7   WebKit                        	       0x132e5ede8 WebKit::WebsiteDataStore::WebsiteDataStore(WTF::Ref<WebKit::WebsiteDataStoreConfiguration, WTF::RawPtrTraits<WebKit::WebsiteDataStoreConfiguration>, WTF::DefaultRefDerefTraits<WebKit::WebsiteDataStoreConfiguration>>&&, PAL::SessionID) + 712 (WebsiteDataStore.cpp:162)
8   WebKit                        	       0x132e5eab8 WebKit::WebsiteDataStore::WebsiteDataStore(WTF::Ref<WebKit::WebsiteDataStoreConfiguration, WTF::RawPtrTraits<WebKit::WebsiteDataStoreConfiguration>, WTF::DefaultRefDerefTraits<WebKit::WebsiteDataStoreConfiguration>>&&, PAL::SessionID) + 48 (WebsiteDataStore.cpp:159)
9   WebKit                        	       0x131ed0f88 void API::Object::constructInWrapper<WebKit::WebsiteDataStore, WTF::Ref<WebKit::WebsiteDataStoreConfiguration, WTF::RawPtrTraits<WebKit::WebsiteDataStoreConfiguration>, WTF::DefaultRefDerefTraits<WebKit::WebsiteDataStoreConfiguration>>, PAL::SessionID&>(id<WKObject>, WTF::Ref<WebKit::WebsiteDataStoreConfiguration, WTF::RawPtrTraits<WebKit::WebsiteDataStoreConfiguration>, WTF::DefaultRefDerefTraits<WebKit::WebsiteDataStoreConfiguration>>&&, PAL::SessionID&) + 84 (APIObject.h:237)
```